src/main/java/it/smartcommunitylab/aac/tos/TosOnAccessFilter.java

```diff
@@ -0,0 +1,160 @@
+import org.springframework.security.web.util.matcher.RequestMatcher;
+import org.springframework.util.CollectionUtils;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.util.WebUtils;
+
+import it.smartcommunitylab.aac.SystemKeys;
+import it.smartcommunitylab.aac.common.NoSuchRealmException;
+import it.smartcommunitylab.aac.common.NoSuchUserException;
+import it.smartcommunitylab.aac.core.MyUserManager;
+import it.smartcommunitylab.aac.core.RealmManager;
+import it.smartcommunitylab.aac.core.auth.ExtendedAuthenticationToken;
+import it.smartcommunitylab.aac.core.auth.RealmAwareAuthenticationEntryPoint;
+import it.smartcommunitylab.aac.core.auth.UserAuthentication;
+import it.smartcommunitylab.aac.model.Realm;
+import it.smartcommunitylab.aac.model.User;
+
+public class TosOnAccessFilter extends OncePerRequestFilter {
+	private final RequestMatcher termsManagedRequestMatcher = new AntPathRequestMatcher("/terms/**");
+	static final String[] SKIP_URLS = { "/api/**", "/html/**", "/js/**", "/lib/**", "/fonts/**", "/italia/**",
+			"/i18n/**" };
+	private final Logger logger = LoggerFactory.getLogger(getClass());
+
+	private final RequestCache requestCache;
+
+	private RedirectStrategy redirectStrategy = new DefaultRedirectStrategy();
+	private RequestMatcher requestMatcher;
+
+	private final RealmManager realmManager;
+	private final MyUserManager userManager;
+
+	public TosOnAccessFilter(RealmManager realmManager, MyUserManager userManager) {
+		// init request cache as store
+		HttpSessionRequestCache cache = new HttpSessionRequestCache();
+		this.requestCache = cache;
+		this.realmManager = realmManager;
+		this.userManager = userManager;
+		this.requestMatcher = buildRequestMatcher();
+	}
+
+	private RequestMatcher buildRequestMatcher() {
+		List<RequestMatcher> antMatchers = Arrays.stream(SKIP_URLS).map(u -> new AntPathRequestMatcher(u))
+				.collect(Collectors.toList());
+
+		return new NegatedRequestMatcher(new OrRequestMatcher(antMatchers));
+
+	}
+
+	public void setRedirectStrategy(RedirectStrategy redirectStrategy) {
+		this.redirectStrategy = redirectStrategy;
+	}
+
+	public void setRequestMatcher(RequestMatcher requestMatcher) {
+		this.requestMatcher = requestMatcher;
+	}
+
+	@Override
+	protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain chain)
+			throws ServletException, IOException {
+		if (requestMatcher.matches(request) && requiresProcessing(request)
+				&& !termsManagedRequestMatcher.matches(request)) {
+			logger.trace("process request for {}", request.getRequestURI());
+			UserAuthentication userAuth = (UserAuthentication) SecurityContextHolder.getContext().getAuthentication();
+			ExtendedAuthenticationToken token = CollectionUtils.firstElement(userAuth.getAuthentications());
+
+			if (token == null) {
+				logger.error("empty token on authentication success");
+				return;
```

Comment:
Do not `return` from filters, because that will stop the processing of the filter chain.
You `return` only if you actually want to stop processing and avoid letting the downstream filters process the request.